components/script/dom/servoparser/async_html.rs

```diff
@@ -318,7 +318,7 @@ impl Tokenizer {
                     let script = self.get_node(&script.id);
                     return TokenizerResult::Script(DomRoot::from_ref(script.downcast().unwrap()));
                 },
-                ToTokenizerMsg::End => unreachable!(),
```

Comment:
This one hasn't been a problem yet, so we can assume it remains unreachable even with the new re-entrant parser.

Reply:
this crashed after I changed the first `_ => unreachable!(),` to `_ => return`

Reply:
Ok, thanks for the info. Then it seems like a good idea indeed.

Reply:
after I only returned with `ToTokenizerMsg::End` and `continue` for others, this is not panicking anymore

Reply:
Thank for confirming. let's just see what @jdm has to say re the comment below.

Reply:
I will try to make time tonight to understand the implications.